events/farrowing_event.py

```diff
@@ -23,48 +28,72 @@ class FarrowingProblem(ModelSQL, ModelView):
     name = fields.Char('Name', required=True, translate=True)
 
 
-#class AbstractEvent(ModelSQL, ModelView, Workflow):
 class FarrowingEvent(AbstractEvent, ImportedEventMixin, ModelSQL, ModelView, Workflow):
     '''Farm Farrowing Event'''
     __name__ = 'farm.farrowing.event'
     _table = 'farm_farrowing_event'
 
     live = fields.Integer('Live', states=_STATES_WRITE_DRAFT,
         depends=_DEPENDS_WRITE_DRAFT)
-    stillborn = fields.Integer('Stillborn', states=_STATES_WRITE_DRAFT,
-        depends=_DEPENDS_WRITE_DRAFT)
-    mummified = fields.Integer('Mummified', states=_STATES_WRITE_DRAFT,
-        depends=_DEPENDS_WRITE_DRAFT)
-    dead = fields.Function(fields.Integer('Dead'),
-        'on_change_with_dead')
+    stillborn = fields.Integer(
+        'Stillborn', states={**_STATES_WRITE_DRAFT,**_INVISIBLE_NOT_GROUP},
+        depends=_DEPENDS_WRITE_DRAFT + ['produced_animal_type'])
+    mummified = fields.Integer(
+        'Mummified', states={**_STATES_WRITE_DRAFT,**_INVISIBLE_NOT_GROUP},
+        depends=_DEPENDS_WRITE_DRAFT + ['produced_animal_type'])
+    dead = fields.Function(fields.Integer('Dead', states={
+            'invisible': ~Equal(Eval('produced_animal_type'), 'group')
+        }), 'on_change_with_dead')
     problem = fields.Many2One('farm.farrowing.problem', 'Problem',
-        states=_STATES_WRITE_DRAFT, depends=_DEPENDS_WRITE_DRAFT)
+        states=_STATES_WRITE_DRAFT,
+        depends=_DEPENDS_WRITE_DRAFT + ['produced_animal_type'])
```

Comment:
produced_animal_type is not necessary here, I think you wanted to add it to "dead"